### PR TITLE
[AMD] catch exception when other processes belong to other users

### PR DESCRIPTION
**Pull request by xw285cornell**
Summary:
It is a long known pain point that if other users are running things, the call of `torch.cuda.memory.list_gpu_processes()` will error out:
```
  torch.cuda.memory.list_gpu_processes()
  File "torch/cuda/memory.py", line 647, in list_gpu_processes
    procs = amdsmi.amdsmi_get_gpu_process_list(handle)  # type: ignore[attr-defined]
  File "amdsmi/py_interface/amdsmi_interface.py", line 1946, in amdsmi_get_gpu_process_list
    _check_res(
  File "amdsmi/py_interface/amdsmi_interface.py", line 510, in _check_res
    raise AmdSmiLibraryException(ret_code)
amdsmi.py_interface.amdsmi_exception.AmdSmiLibraryException: Error code:
	10 | AMDSMI_STATUS_NO_PERM - Permission Denied

```

So just catch this error

Test Plan: torch.cuda.memory.list_gpu_processes() no longer fails

Differential Revision: D59901053

### CODE REVIEW

File: torch/csrc/Module.cpp [hunk 1]
```diff
@@ -702,10 +702,14 @@ PyObject* THPModule_setSDPUseFlash(PyObject* _unused, PyObject* arg) {
   END_HANDLE_TH_ERRORS
 }
 PyObject* THPModule_userEnabledFlashSDP(PyObject* _unused, PyObject* noargs) {
+#ifndef USE_FLASH_ATTENTION
+  Py_RETURN_FALSE;
+#else
```
Comment (eqy) [inline torch/csrc/Module.cpp:707]: Are these changes related to this PR?

**Review by eqy — APPROVED** [review]

File: torch/csrc/Module.cpp (hunk 1 above)
Comment (xw285cornell) [reply torch/csrc/Module.cpp:707]: oops no good catch, I'll remove them

**Review by clee2000 — APPROVED** [review]
